entity-service-impl/src/main/java/org/hypertrace/entity/data/service/EntityNormalizer.java

```diff
@@ -90,7 +90,7 @@ private void verifyMatchingIdentifyingAttributes(String tenantId, Entity request
               "No identifying attributes defined for EntityType: %s", request.getEntityType()));
     }
 
-    if (!idAttrNames.equals(request.getIdentifyingAttributesMap().keySet())) {
+    if (!request.getIdentifyingAttributesMap().keySet().containsAll(idAttrNames)) {
```

Comment:
This could potentially lead to multiple entities being created if an extra attribute(with high cardinality) is passed from the client.

Reply:
If the client is passing an extra identifier, then that's presumably the intent, right?